### PR TITLE
Add the pg_database table and current_database function

### PR DESCRIPTION
**Pull request by mfussenegger**
- [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/general/builtins/scalar.rst [hunk 1]
```diff
@@ -1856,6 +1856,20 @@ Example::
...
+----------------------
+
+The ``current_database`` function returns the name of the current database,
+which in CrateDB will always be ``CrateDB``::
+
+    cr> select current_database();
+    +--------------------+
+    | current_database() |
+    +--------------------+
+    | CrateDB            |
```
Comment (seut) [inline blackbox/docs/general/builtins/scalar.rst]: should we maybe use lower_case instead?

File: sql/src/main/java/io/crate/expression/scalar/CurrentDatabaseFunction.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.types.DataTypes;
+
+import java.util.Collections;
+
+final class CurrentDatabaseFunction extends Scalar<String, Void> {
+    @Override
+    public FunctionInfo info() {
+        return new FunctionInfo(new FunctionIdent("current_database", Collections.emptyList()), DataTypes.STRING);
```
Comment (seut) [inline sql/src/main/java/io/crate/expression/scalar/CurrentDatabaseFunction.java]: why not use a static info?

File: sql/src/main/java/io/crate/metadata/pgcatalog/PgCatalogTableDefinitions.java [hunk 3]
```diff
@@ -54,6 +55,11 @@ public PgCatalogTableDefinitions(InformationSchemaIterables informationSchemaIte
                          || user.hasAnyPrivilege(Privilege.Clazz.VIEW, t.ident().fqn()),
             PgClassTable.expressions()
         ));
+        tableDefinitions.put(PgDatabaseTable.NAME, new StaticTableDefinition<>(
+            () -> singletonList(null),
+            ((user, o) -> true),
```
Comment (seut) [inline sql/src/main/java/io/crate/metadata/pgcatalog/PgCatalogTableDefinitions.java]: we could use the ctor without the predicate argument.

File: blackbox/docs/general/builtins/scalar.rst (hunk 1 above)
Comment (mfussenegger) [reply blackbox/docs/general/builtins/scalar.rst]: /cc @joemoe  any opinion on this?

Comment (joemoe) [reply blackbox/docs/general/builtins/scalar.rst]: in postgres the user can define it, right?

Comment (mfussenegger) [reply blackbox/docs/general/builtins/scalar.rst]: @seut changed it to `crate`

**Pull request merged**